### ISSUE TITLE
non-conformable arguments when running userGWAS

### ISSUE THREAD

**Issue by NicolasC0** [GenomicSEM/GenomicSEM#81]
Hello there!

Thank you very much for developing this package! I'm very new to this type of analysis and I've been trying to run the common factor GWAS on two traits of interest, but I keep running into this error: 

PSYCH_factor <- userGWAS(covstruc = LDSCoutput, SNPs = PSYCH_sumstats, estimation = "DWLS", model = commonfactor.model1, printwarn = TRUE, cores = 30, parallel = TRUE, smooth_check = TRUE, std.lv = TRUE, fix_measurement = TRUE)

[1] "Please note that an update was made to userGWAS on Sept 1 2023  so that the default behavior is to fix the measurement model using the fix_measurement argument."
Error in lav_fit_gfi(WLS.obs = WLS.obs, WLS.est = WLS.est, WLS.V = WLS.V,  :
  non-conformable arguments

My LDSCoutput is as follows, I haven't been able to pinpoint any issues: 
> LDSCoutput
$V
             [,1]         [,2]         [,3]
[1,] 9.243286e-05 2.820133e-05 1.960367e-05
[2,] 2.820133e-05 1.152835e-04 3.987369e-05
[3,] 1.960367e-05 3.987369e-05 4.337307e-04
$S
           trait1        trait2
[1,] 0.20307699 0.09408977
[2,] 0.09408977 0.24121583
$I
          [,1]      [,2]
[1,] 1.0255753 0.2674015
[2,] 0.2674015 1.0080115
$N
         [,1]     [,2]    [,3]
[1,] 89042.57 43652.53 21187.8
$m
[1] 1173569

The model I'm trying to run is very simple, I want to fix the factor loading for both traits to be equal:

commonfactor.model1 <- 'F1 =~ a*trait1+ a*trait2
F1 ~~ F1'
This is my first time dealing with lavaan syntax so I'm not sure this is correct for what I'm looking for. While troubleshooting I freed-up de model to commonfactor.model2 <- 'F1 =~ trait1 + trait2
F1 ~~ F1'
but still encounter the same "non-conformable arguments" error. When I use the commonfactorGWAS function it keeps running for days and never finishes and I'm not sure what the issue is.

Thanks so much for any help!

**Comment by Virginie001** [comment]
has it been solved?

**Comment by XiaoShangSdu** [comment]
问题解决了吗？我的也是一直报错，显示“未整合参数”

**Comment by tuckerdrob** [comment]
non-conformable arguments usually means that you mislabeled a variable
relative to how it is represented in the ldsc object.

**Comment by XiaoShangSdu** [comment]
我的名称检查是一致的
> # LDSC 中的名称
ldsc_names <- colnames(LDSC_11_new$S)
cat("LDSC 名称:", paste(ldsc_names, collapse=", "), "\n")

# 模型中的名称（从模型中提取）
model_names <- c("ADHD", "BIP", "MDD", "ANX", "PTSD", "HL", "HTN", "HYPO", "OB", "NAFLD", "T2D")
cat("模型名称:", paste(model_names, collapse=", "), "\n")

# 检查是否一致
identical(ldsc_names, model_names)
LDSC 名称: ADHD, BIP, MDD, ANX, PTSD, HL, HTN, HYPO, OB, NAFLD, T2D 
模型名称: ADHD, BIP, MDD, ANX, PTSD, HL, HTN, HYPO, OB, NAFLD, T2D 
[1] TRUE
之后重新运行了userGWAS，还是显示未整合参数
> library(GenomicSEM)

load("/data/shangxiao/FinnGen/all_diseases_ldsc/LDSC_11_new.RData")
load("/data/shangxiao/FinnGen/new_sumstats/sumstats_manual_with_MAF.RData")

snps_df <- sumstats_list[[1]][, c("SNP", "A1", "A2", "MAF")]
for(d in names(sumstats_list)) {
  snps_df[[paste0("beta.", d)]] <- sumstats_list[[d]]$BETA
  snps_df[[paste0("se.", d)]] <- sumstats_list[[d]]$SE
}

model_bifactor <- "
  General =~ ADHD + BIP + MDD + ANX + PTSD + HL + HTN + HYPO + OB + NAFLD + T2D
  PsychSpecific =~ ADHD + BIP + MDD + ANX + PTSD
  MetabSpecific =~ HL + HTN + HYPO + OB + NAFLD + T2D
  General ~~ 0*PsychSpecific
  General ~~ 0*MetabSpecific
  PsychSpecific ~~ 0*MetabSpecific
"

test_snps <- snps_df[1:10, ]

result <- userGWAS(
  covstruc = LDSC_11_new,
  SNPs = test_snps,
  model = model_bifactor,
  sub = c("General ~ SNP"),
  cores = 1,
  parallel = FALSE,
  fix_measurement = TRUE,
  smooth_check = TRUE,
  toler = 1e-300,
  estimation = "DWLS"
)

print("Success!")
共有23个警告 (用warnings()来显示)
[1] "Please note that an update was made to userGWAS on Sept 1 2023  so that the default behavior is to fix the measurement model using the fix_measurement argument."
错误于lav_fit_measures(object = object, fit.measures = fit.measures, : 
  非整合参数
